### PR TITLE
fix(decompilation): use cfr lib instead of procyon

### PR DESCRIPTION
**Pull request by rbioteau**
Super types detection was failing in some cases with procyon. Switching to [cfr](https://github.com/leibnitz27/cfr), which seems to be more active, fixes the issue.

Closes STUDIO-4079

### CODE REVIEW

File: plugin/src/test/java/org/bonitasoft/plugin/analyze/ConnectorResolverTest.java [hunk 1]
```diff
@@ -193,6 +193,19 @@ void testJarWithInvalidImplemetationClasses(String testJarFile) throws Exception
         assertEquals(0, implementations.size());
         assertEquals(1, definitions.size());
     }
+    
+    @Test
+    void testCustomConnectos() throws Exception {
```
Comment (laurentleseigneur) [inline plugin/src/test/java/org/bonitasoft/plugin/analyze/ConnectorResolverTest.java:198]:
```suggestion
    void testCustomConnectors() throws Exception {
```
typo

File: plugin/src/main/java/org/bonitasoft/plugin/analyze/ConnectorResolver.java [hunk 2]
```diff
@@ -47,27 +59,21 @@
...
 @Named
 @Singleton
 public class ConnectorResolver {
 
-    public static final String CONNECTOR_TYPE = "org/bonitasoft/engine/connector/Connector";
-    public static final String FILTER_TYPE = "org/bonitasoft/engine/filter/UserFilter";
-    public static final String ABSTRACT_CONNECTOR_TYPE = "org/bonitasoft/engine/connector/AbstractConnector";
-    public static final String ABSTRACT_FILTER_TYPE = "org/bonitasoft/engine/filter/AbstractUserFilter";
+    private static final String CONNECTOR_TYPE = "org.bonitasoft.engine.connector.Connector";
+    private static final String FILTER_TYPE = "org.bonitasoft.engine.filter.UserFilter";
```
Comment (laurentleseigneur) [inline plugin/src/main/java/org/bonitasoft/plugin/analyze/ConnectorResolver.java:68]: not sure to understand this filter type

File: plugin/src/main/java/org/bonitasoft/plugin/analyze/ConnectorResolver.java [hunk 3]
```diff
@@ -47,27 +59,21 @@
...
+    private static final String ABSTRACT_CONNECTOR_TYPE = "org.bonitasoft.engine.connector.AbstractConnector";
+    private static final String ABSTRACT_FILTER_TYPE = "org.bonitasoft.engine.filter.AbstractUserFilter";
 
     private static final Set<String> FILTER_TYPES = Set.of(FILTER_TYPE, ABSTRACT_FILTER_TYPE);
     private static final Set<String> CONNECTOR_TYPES = Set.of(CONNECTOR_TYPE, ABSTRACT_CONNECTOR_TYPE);
```
Comment (laurentleseigneur) [inline plugin/src/main/java/org/bonitasoft/plugin/analyze/ConnectorResolver.java:73]: Shouldn't you also add REST API extension to this list in orger to manage all extension points ?

Comment (rbioteau) [reply plugin/src/main/java/org/bonitasoft/plugin/analyze/ConnectorResolver.java]: Nope

**Pull request merged**